grass/syscall.h

```diff
@@ -13,23 +13,14 @@ struct sys_msg {
     int sender;
     int receiver;
     char content[SYSCALL_MSG_LEN];
+    int received;
```

Comment:
I feel like, instead of `int received`, it is better to do `enum {PENDING, RECEIVED} status;` here.

In other words, instead of checking `received == 1` or not, check `status == RECEIVED` or not.